File: Dictionaries, Lambda and LINQ/Exercises/Phonebook.cs
using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;
using static System.Console;

class Program
{
    static void Main()
    {
        string userInput;
        var PhoneBook = new Dictionary<string, string>();

        while((userInput=ReadLine())!="END")
        {
            char command = userInput.Split()[0][0];

            switch(command)
            {
                case 'A':
                    string name = userInput.Split()[1];
                    string number = userInput.Split()[2];
                    PhoneBook[name] = number;
                    break;
                case 'S':
                    name = userInput.Split()[1];
                    WriteLine(PhoneBook.Keys.Contains(name) ?
                        $"{name} -> {PhoneBook[name]}" :
                        $"Contact {name} does not exist.");
                    break;
            }
        }
    }
}

File: Software Technologies/Java/Exercises/Phonebook.java
package com.company;
import java.util.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import java.util.stream.Collectors;

public class Main {
    static Scanner Input = new Scanner(System.in);

    public static void main(String[] args) {
        Map<String, String> phonebook = new TreeMap<>();

        String commandLine;
        while ((commandLine = Input.nextLine()).compareTo("END")!=0){
            String[] temp = commandLine.split(" ");
            char command = temp[0].charAt(0);
            String name = temp[1];
            String number = temp.length==3? temp[2]:"";

            switch (command){
                case 'A': phonebook.put(name, number); break;
                case 'S':
                    System.out.println(
                            phonebook.containsKey(name)?
                                    name + " -> " + phonebook.get(name):
                                    "Contact " + name + " does not exist."
                    );
                    break;
            }
        }
    }
}